haystack/utils/import_utils.py

```diff
@@ -5,12 +5,54 @@
 import zipfile
 import requests
 import logging
+import importlib
 from pathlib import Path
 
 
 logger = logging.getLogger(__name__)
 
 
+def safe_import(import_path, classname, dep_group):
+    """
+    Method that allows the import of "non-existing" document stores.
```

Comment:
Is also used for other classes besides document stores if I got it correctly

Reply:
Yes, every node that depends on optional dependencies is going to leverage this function at import time.